06-building-a-program-start-to-finish/writing-tests/output/InvertedIndex.spec.js

```diff
@@ -0,0 +1,276 @@
+ () => {
+   it('should update the indexedFiles record when it is invoked', () => {
+     expect(invertedIndex.createIndex(uploads, 'adventureBooks.json')).toEqual({
+       'adventureBooks.json':
+       { a: [true, true, false],
+         alice: [true, false, false],
+         alliance: [false, true, false],
+         an: [false, true, false],
+         and: [true, true, true],
+         battles: [false, false, true],
+         came: [false, false, true],
+         contact: [false, false, true],
+         destroy: [false, true, false],
+         during: [false, false, true],
+         dwarf: [false, true, false],
+         elf: [false, true, false],
+         enters: [true, false, false],
+         falls: [true, false, false],
+         fellowship: [false, true, false],
+         fire: [false, false, true],
+         full: [true, false, false],
+         goblet: [false, false, true],
+         harry: [false, false, true],
+         his: [false, false, true],
+         hobbit: [false, true, false],
+         hole: [true, false, false],
+         imagination: [true, false, false],
+         in: [true, false, true],
+         into: [true, false, false],
+         lord: [false, true, false],
+         man: [false, true, false],
+         of: [true, true, true],
+         persia: [false, false, true],
+         potter: [false, false, true],
+         powerful: [false, true, false],
+         prince: [false, false, true],
+         rabbit: [true, false, false],
+         ring: [false, true, false],
+         rings: [false, true, false],
+         seek: [false, true, false],
+         the: [false, true, true],
+         to: [false, true, false],
+         unusual: [false, true, false],
+         with: [false, false, true],
+         wizard: [false, true, false],
+         wonderland: [true, false, false],
+         world: [true, false, false]
+       }
+     }
+     );
+   });
+
+   it('should return null for fake invertedIndex books format', () => {
+     expect(invertedIndex.createIndex(uploads, 'wrongFormat.json')).toBe(null);
+   });
+   it('should return null for empty file', () => {
+     expect(invertedIndex.createIndex(uploads, 'emptyBook.json')).toBe(null);
+   });
+ });
+
+describe('searchIndex method',
+ () => {
+   let tokens;
+   beforeEach(() => {
+     tokens = ['alice', 'imagination', 'enters', 'powerful', 'usman'];
+   });
+
+   it('should return "object" when its type is checked', () => {
+     expect(typeof (invertedIndex.searchIndex('adventureBooks.json', tokens)))
+     .toBe('object');
+   });
+
+   it('should update the search record when it is invoked', () => {
+     expect(invertedIndex.searchIndex('adventureBooks.json', tokens)).toEqual({
+       'adventureBooks.json': {
+         alice: [true, false, false],
+         enters: [true, false, false],
+         imagination: [true, false, false],
+         powerful: [false, true, false],
+         usman: [false, false, false]
+       }
+     });
+   });
+ });
+
```

Comment:
Are you also handling for the scenario when an invalid "_json_" file is uploaded? I uploaded an invalid "_json_" file and and it threw an error.